portfolio/src/main/webapp/script.js

```diff
@@ -116,3 +124,87 @@ function toggleComments() {
         view.innerText = "Hide Comments";
     }
 }
+
+/**
+ * Retrieves auth info and displays the correct login interface
+ */
+function initializeAuth() {
+  const loginStatusPromise = fetch("/auth");
+
+  loginStatusPromise
+    .then((response) => response.json())
+    .then((loginInfo) => {
+      const loginStatus = loginInfo["loginStatus"];
+      const loginURL = loginInfo["loginURL"];
+      const logoutURL = loginInfo["logoutURL"];
+
+      const authButton = document.getElementById("auth-button");
+      const authGreeting = document.getElementById("auth-greeting");
+
+      if (loginStatus === "true") {
+        authGreeting.innerHTML = "You're logged in:";
+        retrieveComments()
+        authButton.innerHTML = "Log Out";
+        authButton.href = logoutURL;
+        authButton.classList.add("logout");
+        authButton.classList.remove("login");
+      } else {
+        authGreeting.innerHTML = "Please log in here: ";
+
+        authButton.innerHTML = "Log In";
+        authButton.href = loginURL;
+        authButton.classList.add("login");
+        authButton.classList.remove("logout");
+      }
+    });
+}
+
+/**
+ * Hides comment form if user is not signed in. Otherwise display user form.
+ */
+function handleCommentFormVisibility() {
+  const loginStatusPromise = fetch("/auth");
+
+  loginStatusPromise
+    .then((response) => response.json())
+    .then((loginInfo) => {
+      const loginStatus = loginInfo["loginStatus"];
+      const commentForm = document.getElementById("comment-form");
+      const dataDisplay = document.getElementById("data-display");
+      const loginAlert = document.getElementById("comment-login-alert");
+      if (loginStatus === "true") {
+        commentForm.style.visibility = "visible";
+        dataDisplay.styleVisibility = "visible";
+        loginAlert.style.visibility = "collapse";
+      } else {
+        commentForm.style.visibility = "hidden";
+        dataDisplay.styleVisibility = "hidden";
+        loginAlert.style.visibility = "visible";
+      }
+    });
+}
+
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+/** Creates a chart and adds it to the page. */
+function drawChart() {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Animal');
+  data.addColumn('number', 'Count');
+        data.addRows([
+          ['Lions', 10],
+          ['Tigers', 5],
+          ['Bears', 15]
```

Comment:
General comment: Make sure you are adding/using APIs that are useful to your portfolio, not just checking boxes off the task list. Look for APIs and enhancements that will make your page a better representation of you!

Reply:
I changed this to a poll about quarantine activities in another pull request, I dont think it represents me, but I'm interested in the data that would come out of it so I think it fits well. I might even add more polls later